File: vue/src/api/key.js
import request from '@/utils/request'

export function fetchList() {
    return request({
        url: '/keys/list',
        method: 'post'
    })
}


export function addKey(data) {
    return request({
        url: '/keys/add',
        method: 'post',
        data
    })
}

export function delKey(id) {
    return request({
        url: "/keys/delete",
        method: 'get',
        params: { id }
    })
}


export function checkName(name) {
    return request({
        url: '/keys/check/name',
        method: 'get',
        params: { name }
    })
}


export function getMykeys() {
    return request({
        url: '/keys/get/me',
        method: 'post'
    })
}

File: go/routegroup/document/key.go
package document

import (
	"itflow/handle"
	"itflow/model"

	"github.com/hyahm/xmux"
)

var Key *xmux.RouteGroup

func init() {
	Key = xmux.NewRouteGroup()
	Key.Post("/keys/list", handle.KeyList)
	Key.Post("/keys/add", handle.AddKey).BindJson(&model.Auth{})
	Key.Get("/keys/delete", handle.DeleteKey)
	Key.Get("/keys/check/name", handle.CheckKeyName)
	Key.Post("/keys/get/me", handle.GetMykeys)
}
